src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/CallbackValidations.java

```diff
@@ -283,6 +288,40 @@ static boolean hasOcr(CaseDetails theCase) {
         return valid(classification);
     }
 
+    public static Validation<String, Classification> hasJourneyClassificationForAttachToCase(CaseDetails theCase) {
+        Optional<String> classificationOption = getJourneyClassification(theCase);
+
+        return classificationOption
+            .map(classification -> Try.of(() -> Classification.valueOf(classification)))
+            .map(Try::toValidation)
+            .map(validation -> validation
+                .mapError(throwable -> "Invalid journeyClassification. Error: " + throwable.getMessage())
+                .flatMap(classification -> validateClassificationForAttachToCase(classification, theCase))
+            )
+            .orElse(invalid("Missing journeyClassification"));
+    }
+
+    private static Validation<String, Classification> validateClassificationForAttachToCase(
+        Classification classification,
+        CaseDetails theCase
+    ) {
+        if (!VALID_CLASSIFICATIONS_FOR_ATTACH_TO_CASE.contains(classification)) {
+            return invalid(format(
+                "The current journey classification %s is not allowed for attaching to case",
+                classification
+            ));
+        }
+
+        if (SUPPLEMENTARY_EVIDENCE_WITH_OCR.equals(classification) && !hasOcr(theCase)) {
+            return invalid(format(
+                "The current journey classification %s is not allowed without OCR",
+                classification
+            ));
+        }
+
+        return valid(classification);
+    }
+
```

Comment:
Do we really need to add **any** changes to this file? it is enough to just use `hasJourneyClassification` and everything else is/will be/should be done in service (check previous suggestion). Validation on classification has been already done at the top level before executing any activity on callback. First mandatory fail fast check ;)

Reply:
Will do